pages/linux/nl.md

```diff
@@ -0,0 +1,24 @@
+# nl
+
+> A utility for numbering lines, either from a file, or from standard oinput.
+
+- Number lines in a file:
+
+`nl {{source_file}}`
+
+- Number lines where there is printable text only:
+
+`nl -t {{source_file}}`
+
+- Number lines that match a regular expression (Starting with a Capital 
+A is Exampled):
+
+`nl -b p^A {{source_file}}`
+
+- Can be useful as an alternative to grep {{source_file}} -n 
+{{search_term}}:
+
+`nl ./file | grep match` 
```

Comment:
Change ./file to use tokens too. Although I doubt how useful this example is because nl {{file}} is already in the first example, and this one just uses another command at the end.